src/jit/compiler.cpp

```diff
@@ -2716,8 +2713,11 @@ void Compiler::compSetProcessor()
             codeGen->getEmitter()->SetContainsAVX(false);
             codeGen->getEmitter()->SetContains256bitAVX(false);
         }
-        else if (CanUseSSE4())
+        else if (compSupports(InstructionSet_SSSE3) || compSupports(InstructionSet_SSE41) || compSupports(InstructionSet_SSE42))
         {
+            // Emitter::UseSSE4 controls whether we support the 4-byte encoding for certain
+            // instructions. We need to check if either is supported independently, since
+            // it is currently possible to enable/disable them separately.
             codeGen->getEmitter()->SetUseSSE4(true);
```

Comment:
We might want to consider changing the name of this flag, as this is a bit confusing. But I don't think it's critical.

Reply:
I agree. I'll log a bug to track it.

Reply:
https://github.com/dotnet/coreclr/issues/16421